### ISSUE TITLE
Capitalized variables (i.e starting with an uppercase letter, e.g. Foo) accepted in for loops

### ISSUE THREAD

**Issue by dputhier** [PyCQA/pylint#2695]
## Question

Dear all,
I would like to understand why Capitalized variables (i.e starting with an uppercase letter and followed by lowercases, e.g. Foo)  are accepted inside for  loops. In the example below, Foocap doesn't conform to UPPER_CASE as expected (as it is considered to be a constant) but Foocapfor is not detected as non-compliant (while it does not conform to UPPERCASE or snakecase). 
Best 

	'''
	Nothing important
	'''

	Foocap = 2

	for i in range(10):
	    Foocapfor = 2

Which is evaluated to 

	pylint --py3k --enable=all  test_4.py                                                          ✔  10412  10:25:13
	************* Module test_4
	test_4.py:5:0: C0103: Constant name "Foocap" doesn't conform to UPPER_CASE naming style (invalid-name)

**Comment by PCManticore** [comment]
Thanks, I can reproduce the bug.

**Comment by brycepg** [comment]
So looking at `pylint/checkers/base.py` If I change 1741 to remove the `not in_loop` check we'd have this change

But is a variable in set in a for loop really a constant?

```
1740 	        elif isinstance(frame, astroid.Module):
1741 	            if isinstance(assign_type, astroid.Assign) and not in_loop(assign_type):
1742 	                if isinstance(utils.safe_infer(assign_type.value), astroid.ClassDef):
1743 	                    self._check_name("class", node.name, node)
1744 	                else:
1745 	                    if not _redefines_import(node):
1746 	                        # Don't emit if the name redefines an import
1747 	                        # in an ImportError except handler.
1748 ->	                        self._check_name("const", node.name, node)
```

**Comment by PCManticore** [comment]
Yes, I don't think it's really a constant, but that line assumes `pylint`'s definition of a constant, which is any variable at the module level. Removing the `for_loop` check sounds good to me.
